tracker.py

```diff
@@ -0,0 +1,24 @@
+"""Module for base tracker."""
+
+from typing import Generic, Optional, TypeVar
+
+T = TypeVar("T")
+
+class Tracker(Generic[T]):
```

Comment:
How about moving `notify_callback()` functions into `Tracker`? Their first arguments are the `Tracker` objects, so it would be fine to make them an abstract method of `Tracker`.
Then we don't have to use `functools.partial()`.

Reply:
It looks much better! I will apply it.